support-frontend/assets/helpers/redux/checkout/checkoutSelectors.ts

```diff
@@ -0,0 +1,73 @@
+import type { ContributionsState } from '../contributionsStore';
+import { getContributionType } from './product/selectors/productType';
+
+function getPaymentMethodErrors(
+	state: ContributionsState,
+): Record<string, string[]> {
+	const { payment } = state.page.checkoutForm;
+
+	switch (payment.paymentMethod) {
+		case 'Stripe':
+			return payment.stripe.errors;
+
+		case 'DirectDebit':
+			return payment.directDebit.errors ?? {};
+
+		case 'Sepa':
+			return payment.sepa.errors;
+
+		default:
+			return {};
+	}
+}
+
+function formHasPaymentMethodErrors(state: ContributionsState): boolean {
+	const errors = getPaymentMethodErrors(state);
+	const errorObjectHasKeys = !!Object.keys(errors).length;
+	const atLeastOneKeyHasErrors = Object.keys(errors).some(
```

Comment:
If `errorObjectHasKeys` is false we can return `false` without needing to check whether  `atLeastOneKeyHasErrors`